internal/utils/config/controlplane.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/samber/lo"
+	corev1 "k8s.io/api/core/v1"
+
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
+
+var kicInKonnectDefaults = func(secretName string) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  "CONTROLLER_KONNECT_ADDRESS",
+			Value: "<KONNECT-ADDRESS>",
+		},
+		{
+			Name:  "CONTROLLER_KONNECT_CONTROL_PLANE_ID",
+			Value: "<KONNECT-CONTROL-PLANE-ID>",
+		},
+		{
+			Name:  "CONTROLLER_KONNECT_LICENSING_ENABLED",
+			Value: "true",
+		},
+		{
+			Name:  "CONTROLLER_KONNECT_SYNC_ENABLED",
+			Value: "true",
+		},
+		{
+			Name:  "CONTROLLER_FEATURE_GATES",
+			Value: "FillIDs=true",
+		},
+		{
+			Name: "CONTROLLER_KONNECT_TLS_CLIENT_KEY",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: "tls.key",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: secretName,
+					},
+				},
+			},
+		},
+		{
+			Name: "CONTROLLER_KONNECT_TLS_CLIENT_CERT",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: "tls.crt",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: secretName,
+					},
+				},
+			},
+		},
+	}
+}
+
+// KongInKonnectDefaults returns the slice of Konnect-related env vars properly configured.
+func KICInKonnectDefaults(konnectExtensionStatus konnectv1alpha1.KonnectExtensionStatus) ([]corev1.EnvVar, error) {
+	var template []corev1.EnvVar
+	switch konnectExtensionStatus.Konnect.ClusterType {
+	case konnectv1alpha1.ClusterTypeK8sIngressController:
+		template = kicInKonnectDefaults(konnectExtensionStatus.DataPlaneClientAuth.CertificateSecretRef.Name)
+	case konnectv1alpha1.ClusterTypeControlPlane:
+		return nil, fmt.Errorf("unsupported Konnect cluster type: %s", konnectExtensionStatus.Konnect.ClusterType)
+	default:
+		// default never happens as the validation is at the CRD level
+		panic(fmt.Sprintf("unsupported Konnect cluster type: %s", konnectExtensionStatus.Konnect.ClusterType))
+	}
+
+	newEnvSet := make([]corev1.EnvVar, len(template))
+	for i, env := range template {
+		newValue := strings.ReplaceAll(env.Value, "<KONNECT-ADDRESS>", buildKonnectAddress(konnectExtensionStatus.Konnect.Endpoints.ControlPlaneEndpoint))
+		newValue = strings.ReplaceAll(newValue, "<KONNECT-CONTROL-PLANE-ID>", konnectExtensionStatus.Konnect.ControlPlaneID)
+		env.Value = newValue
+		newEnvSet[i] = env
+	}
```

Comment:
Nit: rather than using the "template" slice, why don't we just use a function which will take 2 parameters: konnectAddress and CP ID and which will return the whole slice prepared for us. WDYT?

Reply:
Refactored in a0be20c8cc867dee4dfb4b8df507be1e9661e126